numpy/core/src/umath/override.c

```diff
@@ -0,0 +1,479 @@
+                }
+            }
+            PyDict_SetItemString(*normal_kwds, "out", obj);
+            Py_DECREF(obj);
+        }
+    }
+    return 0;
+}
+
+static int
+normalize_reduce_accumulate_args(PyUFuncObject *ufunc, PyObject *args,
+                                 PyObject **normal_args, PyObject **normal_kwds)
+{
+    /*
+     * ufunc.reduce(a[, axis, dtype, out, keepdims])
+     * ufunc.accumulate(a[, axis, dtype, out])
+     * the number of arguments has been checked in PyUFunc_GenericReduction.
+     */
+    int nargs = PyTuple_GET_SIZE(args);
+    int i;
+    PyObject *obj;
+    static char *kwlist[] = {"array", "axis", "dtype", "out", "keepdims"};
+
+    *normal_args = PyTuple_GetSlice(args, 0, 1);
+    if (*normal_args == NULL) {
+        return -1;
+    }
+
+    for (i = 1; i < nargs; i++) {
+        if (PyDict_GetItemString(*normal_kwds, kwlist[i])) {
+            PyErr_Format(PyExc_TypeError,
+                         "argument given by name ('%s') and position (%d)",
+                         kwlist[i], i);
+            return -1;
+        }
+        obj = PyTuple_GET_ITEM(args, i);
+        if (obj != Py_None) {
+            if (i == 3) {
+                obj = PyTuple_GetSlice(args, 3, 4);
+            }
+            PyDict_SetItemString(*normal_kwds, kwlist[i], obj);
+            if (i == 3) {
+                Py_DECREF(obj);
+            }
+        }
+    }
+    return 0;
+}
+
+static int
+normalize_reduceat_args(PyUFuncObject *ufunc, PyObject *args,
+                    PyObject **normal_args, PyObject **normal_kwds)
+{
+    /*
+     * ufunc.reduceat(a, indicies[, axis, dtype, out])
+     * the number of arguments has been checked in PyUFunc_GenericReduction.
+     */
+    int i;
+    int nargs = PyTuple_GET_SIZE(args);
+    PyObject *obj;
+    static char *kwlist[] = {"array", "indices", "axis", "dtype", "out"};
+
+    /* a and indicies */
+    *normal_args = PyTuple_GetSlice(args, 0, 2);
+    if (*normal_args == NULL) {
+        return -1;
+    }
+
+    for (i = 2; i < nargs; i++) {
+        if (PyDict_GetItemString(*normal_kwds, kwlist[i])) {
+            PyErr_Format(PyExc_TypeError,
+                         "argument given by name ('%s') and position (%d)",
+                         kwlist[i], i);
+            return -1;
+        }
+        obj = PyTuple_GET_ITEM(args, i);
+        if (obj != Py_None) {
+            if (i == 4) {
+                obj = PyTuple_GetSlice(args, 4, 5);
+            }
+            PyDict_SetItemString(*normal_kwds, kwlist[i], obj);
+            if (i == 4) {
+                Py_DECREF(obj);
+            }
+        }
+    }
+    return 0;
+}
+
+static int
+normalize_outer_args(PyUFuncObject *ufunc, PyObject *args,
+                    PyObject **normal_args, PyObject **normal_kwds)
+{
+    /*
+     * ufunc.outer(A, B)
+     * This has no kwds so we don't need to do any kwd stuff.
+     */
+    *normal_args = PyTuple_GetSlice(args, 0, 2);
+    return (*normal_args == NULL);
+}
```

Comment:
I had left this as it was when I found it, but you're right, the two should just be combined. I have now done that.